www/src/components/navigation/ColorSchemeSelect.tsx

```diff
@@ -0,0 +1,141 @@
+import { Listbox, Transition } from "@headlessui/react";
+import { Fragment, useLayoutEffect, useState } from "react";
+
+type ThemesNames = (typeof themesOptions)[number]["name"];
+
+const findThemeByName = (name: ThemesNames) => {
+  return themesOptions.find((theme) => theme.name === name);
+};
+
+const themesOptions = [
+  {
+    name: "system",
+    Icon: (
+      <svg
+        xmlns="http://www.w3.org/2000/svg"
+        className="h-6 w-6 fill-current stroke-current transition-colors duration-300"
+        viewBox="0 0 24 24"
+      >
+        <path
+          d="m11.998 2c5.517 0 9.997 4.48 9.997 9.998 0 5.517-4.48 9.997-9.997 9.997-5.518 0-9.998-4.48-9.998-9.997 0-5.518 4.48-9.998 9.998-9.998zm0 1.5c-4.69 0-8.498 3.808-8.498 8.498s3.808 8.497 8.498 8.497z"
+          fillRule="nonzero"
+        />
+      </svg>
+    ),
+    clickHandler: () => {
+      localStorage.setItem("theme", "");
+    },
+  },
+  {
+    name: "light",
+    Icon: (
+      <svg
+        xmlns="http://www.w3.org/2000/svg"
+        className="h-6 w-6 fill-current stroke-current stroke-2 transition-colors duration-300"
+        viewBox="0 0 24 24"
+      >
+        <path
+          strokeLinecap="round"
+          strokeLinejoin="round"
+          d="M12 3v2.25m6.364.386l-1.591 1.591M21 12h-2.25m-.386 6.364l-1.591-1.591M12 18.75V21m-4.773-4.227l-1.591 1.591M5.25 12H3m4.227-4.773L5.636 5.636M15.75 12a3.75 3.75 0 11-7.5 0 3.75 3.75 0 017.5 0z"
+        />
+      </svg>
+    ),
+    clickHandler: () => {
+      document.documentElement.classList.remove("dark");
+      document.documentElement.classList.add("light");
+      localStorage.setItem("theme", "light");
+    },
+  },
+  {
+    name: "dark",
+    Icon: (
+      <svg
+        xmlns="http://www.w3.org/2000/svg"
+        className="h-6 w-6 fill-current stroke-current transition-colors duration-300"
+        viewBox="0 0 20 20"
+      >
+        <path d="M17.293 13.293A8 8 0 016.707 2.707a8.001 8.001 0 1010.586 10.586z" />
+      </svg>
+    ),
+    clickHandler: () => {
+      document.documentElement.classList.add("dark");
+      document.documentElement.classList.remove("light");
+      localStorage.setItem("theme", "dark");
+    },
+  },
+] as const;
+
+export default function ColorSchemeSelect() {
+  const [selectedTheme, setSelectedTheme] = useState<ThemesNames | null>(null);
+
+  useLayoutEffect(() => {
+    const theme = localStorage.getItem("theme") as ThemesNames;
+    setSelectedTheme(theme || "system");
+
+    window
+      .matchMedia("(prefers-color-scheme: dark)")
+      .addEventListener("change", (e) => {
+        if (e.matches) {
+          findThemeByName("dark")?.clickHandler();
+          setSelectedTheme("dark");
+        } else {
+          setSelectedTheme("light");
+          findThemeByName("light")?.clickHandler();
+        }
+      });
+  }, []);
```

Comment:
this event listener should only run when the system option is selected, we should also have a cleanup function